### ISSUE TITLE
Dear sebres or someone, help me bros.. fail2ban won't ban anything at CentOS 7.6

### ISSUE THREAD

**Issue by xtronaltic** [fail2ban/fail2ban#2380]
(First of all, I'm stilling learning linux. I wouldn't ask for help unless I really don't know how to solve it myself. Sorry for bothering you guys)

The issue I have here is fail2ban-0.9.7 just won't ban anything at CentOS Linux release 7.6.1810 (Core). I didn't install iptables, just using the default firewalld. I also disabled SElinux.

Here're the things I did:

1. `yum -y install epel-release.noarch`
2. `yum install fail2ban`
3. `vi /etc/fail2ban/jail.local`
4. > #I typed all these below to the new  /etc/fail2ban/jail.local file.
```ini
[DEFAULT]
ignoreip = 127.0.0.1/8
#In order to test fail2ban myself, I set bantime to 60
bantime = 60
findtime = 300
maxretry = 3
#I want to use firewalld instead of iptables
banaction = firewallcmd-ipset
#I don't need fail2ban to send me emails, just want it to ban ips.
action = %(action_)s
[sshd]
enabled = true
filter = sshd
port = ssh
action = %(action_)s
logpath = /var/log/secure
[sshd-ddos]
enabled = true
filter = sshd-ddos
port = ssh
action = %(action_)s
logpath = /var/log/secure
```
>
5. `:wq`
6. `systemctl enable fail2ban`
7. `systemctl start fail2ban`
8. > #Because I also want to ban the ips which show up at /var/log/secure with the warning of "Did not receive identification string from x.x.x.x". So I followed sebres' suggestion from https://github.com/fail2ban/fail2ban/issues/1284. I copied `filter.d/sshd.conf` into `filter.d/sshd.local` and add following to the failregex in `filter.d/sshd.local`:
             `^%(__prefix_line)sDid not receive identification string from <HOST>\s*$` 
#Please note that I'm using fail2ban-0.9.7. I saw similar line like above from `/etc/fail2ban/filter.d/sshd-ddos.conf` and I do use jail `[sshd-ddos]` at `/etc/fail2ban/jail.local`.  So I'm also a bit confused regarding the change sebres suggested from https://github.com/fail2ban/fail2ban/issues/1284 to `filter.d/sshd.local`.
9.  `systemctl restart fail2ban`
10. `tail /var/log/fail2ban.log`
11. 
```
 fail2ban.jail           : INFO    Jail 'sshd-ddos' uses systemd {}
 fail2ban.jail           : INFO    Initiated 'systemd' backend
 fail2ban.filter        : INFO    Set maxRetry = 3
 fail2ban.filter        : INFO    Set jail log file encoding to UTF-8
 fail2ban.actions        : INFO    Set banTime = 60
 fail2ban.filter         : INFO    Set findtime = 300
 fail2ban.filter         : INFO    Set maxlines = 10
 fail2ban.filtersystemd  : INFO    Added journal match for: '_SYSTEMD_UNIT=sshd.service + _COMM=sshd'
 fail2ban.jail           : INFO    Jail 'sshd' started
 fail2ban.jail           : INFO    Jail 'sshd-ddos' started
```
12. >#Disconected. Testing fail2ban by mistyping password 5 times, although in `/etc/fail2ban/jail.local` the `maxretry = 3`.  Immediately after that I still can login to my server with the right password. I see the warning regarding "several failed attempts..." when I try to login to root user.
13. `fail2ban-client status sshd`
14. > Status for the jail: sshd
|- Filter
|  |- Currently failed:	0
|  |- Total failed:	0
|  `- Journal matches:	_SYSTEMD_UNIT=sshd.service + _COMM=sshd
`- Actions
   |- Currently banned:	0
   |- Total banned:	0
   `- Banned IP list:	
15. `grep "Failed" /var/log/secure`
16. 
```
Failed password for steve from x.x.x.x     
Failed password for steve from x.x.x.x     
Failed password for steve from x.x.x.x     
Failed password for steve from x.x.x.x     
Failed password for steve from x.x.x.x     
```
Any ideas what I did wrong? I don't know whether it's CentOS 7/Firewalld related issue or fail2ban config issue. 

Please help, @sebres

**Comment by sebres** [comment]
Does really this log-lines in /var/log/secure looks exactly so?
```
Failed password for steve from x.x.x.x
```
So completely without timestamp etc?

**Comment by xtronaltic** [comment]
> Does really this log-lines in /var/log/secure looks exactly so?
> 
> ```
> Failed password for steve from x.x.x.x
> ```
> So completely without timestamp etc?

No sebres, I just delete it to make it short. It has timestamp and everything. 

`Failed password for steve from x.x.x.x`

This is me testing fail2ban by mistyping password after setting up fail2ban. I pulled it from /var/log/secure.

**Comment by sebres** [comment]
> I just delete it to make it short. It has timestamp and everything.

Then provide the excerpt **with full log-line** (you could blacken your sensitive data like hostname with comparable values, so using the same characters, e. g. replace "yourhostname" with "server" or "host", etc).
Otherwise we'd not help you here.

**Comment by xtronaltic** [comment]
My fault @sebres . 

I normaly use SSH Key to login. Because I needed to test fail2ban, I enabled `PasswordAuthentication` and then tested fail2ban by typing wrong password in Xshell. After trying 6 times, I use my SSH key to login and see whether fail2ban is working or not.

**Here's  `/var/log/secure` full log-line:**

```
Mar 25 17:52:04 SERVER sshd[****]: Did not receive identification string from 127.0.0.1 port *****
Mar 25 17:52:30 SERVER sshd[****]: Did not receive identification string from 127.0.0.1 port *****
Mar 25 17:52:38 SERVER sshd[****]: Did not receive identification string from 127.0.0.1 port *****
Mar 25 17:52:44 SERVER sshd[****]: Did not receive identification string from 127.0.0.1 port *****
Mar 25 17:52:49 SERVER sshd[****]: Did not receive identification string from 127.0.0.1 port *****
Mar 25 17:52:55 SERVER sshd[****]: Did not receive identification string from 127.0.0.1 port *****
Mar 25 17:54:06 SERVER sshd[****]: Did not receive identification string from 127.0.0.1 port *****
Mar 25 17:54:12 SERVER sshd[****]: Did not receive identification string from 127.0.0.1 port *****
Mar 25 17:54:20 SERVER sshd[****]: Did not receive identification string from 127.0.0.1 port *****
Mar 25 18:09:15 SERVER sshd[****]: pam_unix(sshd:session): session closed for user USER
Mar 25 18:09:15 SERVER su: pam_unix(su:session): session closed for user root
Mar 25 18:09:28 SERVER sshd[****]: pam_unix(sshd:auth): authentication failure; logname= uid=0 euid=0 tty=ssh ruser= rhost=MYIP.MYIP.MYIP.MYIP  user=USER
Mar 25 18:09:29 SERVER sshd[****]: Failed password for USER from MYIP.MYIP.MYIP.MYIP port ***** ssh2
Mar 25 18:09:33 SERVER sshd[****]: Failed password for USER from MYIP.MYIP.MYIP.MYIP port ***** ssh2
Mar 25 18:09:36 SERVER sshd[****]: Failed password for USER from MYIP.MYIP.MYIP.MYIP port ***** ssh2
Mar 25 18:09:39 SERVER sshd[****]: Failed password for USER from MYIP.MYIP.MYIP.MYIP port ***** ssh2
Mar 25 18:09:41 SERVER sshd[****]: Failed password for USER from MYIP.MYIP.MYIP.MYIP port ***** ssh2
Mar 25 18:09:41 SERVER sshd[****]: error: maximum authentication attempts exceeded for USER from MYIP.MYIP.MYIP.MYIP port ***** ssh2 [preauth]
Mar 25 18:09:41 SERVER sshd[****]: Disconnecting: Too many authentication failures [preauth]
Mar 25 18:09:41 SERVER sshd[****]: PAM 4 more authentication failures; logname= uid=0 euid=0 tty=ssh ruser= rhost=MYIP.MYIP.MYIP.MYIP  user=USER
Mar 25 18:09:41 SERVER sshd[****]: PAM service(sshd) ignoring max retries; 5 > 3
Mar 25 18:09:45 SERVER sshd[****]: pam_unix(sshd:auth): authentication failure; logname= uid=0 euid=0 tty=ssh ruser= rhost=MYIP.MYIP.MYIP.MYIP  user=USER
Mar 25 18:09:46 SERVER sshd[****]: Failed password for USER from MYIP.MYIP.MYIP.MYIP port ***** ssh2
Mar 25 18:09:47 SERVER sshd[****]: error: Received disconnect from MYIP.MYIP.MYIP.MYIP port *****:0:  [preauth]
Mar 25 18:09:47 SERVER sshd[****]: Disconnected from MYIP.MYIP.MYIP.MYIP port ***** [preauth]
Mar 25 18:09:59 SERVER sshd[****]: Accepted publickey for USER from MYIP.MYIP.MYIP.MYIP port ***** ssh2: RSA SHA256:*************************************************************
Mar 25 18:09:59 SERVER sshd[****]: pam_unix(sshd:session): session opened for user USER by (uid=0)
Mar 25 18:10:26 SERVER su: pam_unix(su:session): session opened for user root by USER(uid=1000)
```
**Here's `/var/log/fail2ban.log`**

```
2019-03-25 17:34:42,801 fail2ban.jail           [****]: INFO    Jail 'sshd-ddos' uses systemd {}
2019-03-25 17:34:42,802 fail2ban.jail           [****]: INFO    Initiated 'systemd' backend
2019-03-25 17:34:42,803 fail2ban.filter         [****]: INFO    Set maxRetry = 3
2019-03-25 17:34:42,804 fail2ban.filter         [****]: INFO    Set jail log file encoding to UTF-8
2019-03-25 17:34:42,804 fail2ban.actions        [****]: INFO    Set banTime = 60
2019-03-25 17:34:42,804 fail2ban.filter         [****]: INFO    Set findtime = 300
2019-03-25 17:34:42,805 fail2ban.filter         [****]: INFO    Set maxlines = 10
2019-03-25 17:34:42,820 fail2ban.filtersystemd  [****]: INFO    Added journal match for: '_SYSTEMD_UNIT=sshd.service + _COMM=sshd'
2019-03-25 17:34:42,832 fail2ban.jail           [****]: INFO    Jail 'sshd' started
2019-03-25 17:34:42,850 fail2ban.jail           [****]: INFO    Jail 'sshd-ddos' started
```

**Comment by xtronaltic** [comment]
**Here's the `fail2ban-regex --print-all-matched /var/log/secure /etc/fail2ban/jail.local` result:**


```
Running tests
=============

Use   failregex file : /etc/fail2ban/jail.local
Use         log file : /var/log/secure
Use         encoding : UTF-8


Results
=======

Failregex: 0 total

Ignoreregex: 0 total

Date template hits:
|- [# of hits] date format
|  [193] (?:DAY )?MON Day 24hour:Minute:Second(?:\.Microseconds)?(?: Year)?
`-

Lines: 193 lines, 0 ignored, 0 matched, 193 missed
[processed in 0.01 sec]

Missed line(s): too many to print.  Use --print-all-missed to print all 193 lines
```

**Comment by sebres** [comment]
So, trying this with current sshd-filter (0.10 and 0.9) show me:
```
fail2ban-regex 'Mar 25 18:09:29 SERVER sshd[1234]: Failed password for USER from 192.0.2.1 port 12345 ssh2' sshd
```
<details>
<summary>output of 0.9</summary>

```
Running tests
=============

Use   failregex filter file : sshd, basedir: /Projects/fail2ban-0.9/config
Use         maxlines : 10
Use      single line : Mar 25 18:09:29 SERVER sshd[1234]: Failed password...


Results
=======

Failregex: 1 total
|-  #) [# of hits] regular expression
|   3) [1] ^(?:\[\])?\s*(?:<[^.]+\.[^.]+>\s+)?(?:\S+\s+)?(?:kernel: \[ *\d+\.\d+\]\s+)?(?:@vserver_\S+\s+)?(?:(?:(?:\[\d+\])?:\s+[\[\(]?sshd(?:\(\S+\))?[\]\)]?:?|[\[\(]?sshd(?:\(\S+\))?[\]\)]?:?(?:\[\d+\])?:?)\s+)?(?:\[ID \d+ \S+\]\s+)?(?:(?:error|fatal): (?:PAM: )?)?Failed \S+ for (?P<cond_inv>invalid user )?(?P<user>(?P<cond_user>\S+)|(?(cond_inv)(?:(?! from ).)*?|[^:]+)) from <HOST>(?: port \d+)?(?: on \S+(?: port \d+)?)?(?: ssh\d*)?(?(cond_user): |(?:(?:(?! from ).)*)$)
`-

Ignoreregex: 0 total

Date template hits:
|- [# of hits] date format
|  [1] (?:DAY )?MON Day 24hour:Minute:Second(?:\.Microseconds)?(?: Year)?
`-

Lines: 1 lines, 0 ignored, 1 matched, 0 missed
[processed in 0.00 sec]
```
</details>

<details>
<summary>output of 0.10</summary>

```
Running tests
=============

Use   failregex filter file : sshd, basedir: /Projects/fail2ban.master/config/
Use         maxlines : 1
Use      datepattern : Default Detectors
Use      single line : Mar 25 18:09:29 SERVER sshd[1234]: Failed password...


Results
=======

Failregex: 1 total
|-  #) [# of hits] regular expression
|   4) [1] ^Failed \b(?!publickey)\S+ for (?P<cond_inv>invalid user )?<F-USER>(?P<cond_user>\S+)|(?(cond_inv)(?:(?! from ).)*?|[^:]+)</F-USER> from <HOST>(?: (?:port \d+|on \S+)){0,2}(?: ssh\d*)?(?(cond_user): |(?:(?:(?! from ).)*)$)
`-

Ignoreregex: 0 total

Date template hits:
|- [# of hits] date format
|  [1] {^LN-BEG}(?:DAY )?MON Day %k:Minute:Second(?:\.Microseconds)?(?: ExYear)?
`-

Lines: 1 lines, 0 ignored, 1 matched, 0 missed
[processed in 0.00 sec]
```
</details>

Because there are matches in both versions, I'd suggest - update your filter.
For 0.9 version you can find it here - https://github.com/fail2ban/fail2ban/blob/0.9/config/filter.d/sshd.conf
*(Click on "Raw" to obtain raw copy)*

**Comment by sebres** [comment]
By the way, testing via `fail2ban-regex ... /etc/fail2ban/jail.local` is wrong
You should use filter file (or expression) instead of `jail.local`.
fail2ban-regex operates with filter/regex and knows nothing about jails.

**Comment by xtronaltic** [comment]
Thank you, @sebres . I'm still learning CentOS. Please ignore my mistakes. 

You're suggesting my current fail2ban 0.9.7 sshd-filter is not working in CentOS 7.6, right?

I'll copy the code from https://github.com/fail2ban/fail2ban/blob/0.9/config/filter.d/sshd.conf and update `filter.d/sshd.local`.

By the way, if I want to use fail2ban to ban ips which are related to `Did not receive identification string from`, after updating `filter.d/sshd.local`, what should I do next? Should I update `filter.d/sshd-ddos.conf` as well from https://github.com/fail2ban/fail2ban/blob/0.9/config/filter.d?

**Comment by sebres** [comment]
You could update all filters (e. g. download the version as zip). 
Just I have basically no idea that all filters would be CentOS compatible (I don't know which changes exactly were done by the maintainers of your distribution).
So try it yourself.

**Comment by xtronaltic** [comment]
Thank you @sebres for your time and help. I`ll try to update all filters.

I tested again, same way, new `sshd.local` filter was working! But I still could login to my server and fail2ban showed my ip was banned. Firewalld didn't block my access.

Is there anything wrong between fail2ban 0.9.7 and Firewalld in CentOS 7.6?

**Comment by sebres** [comment]
> Firewalld didn't block my access.

If you see proper entry in firewalld rules, then it is rather an firewalld issue (or possibly some misconfiguration like some rules before fail2ban rules, like FORWARD before INPUT rules of fail2ban).
So you have to check all chains are correct configured.

If not, check fail2ban.log for some errors after an entry `[sshd] Ban x.x.x.x`.

**Comment by xtronaltic** [comment]
Thanks again @sebres ! You helped a lot and I also learned a lot from you, brother! I checked my OS, I didn't find anything unusual. But I tried this method https://github.com/fail2ban/fail2ban/issues/1474#issuecomment-272659488 and finally I could ban myself by fail2ban:

`vi /etc/fail2ban/action.d/custom-firewalld.conf`
```
[INCLUDES]
before  =

[Definition]
actionstart =
actionstop =
actioncheck =

actionflush = sed -i '/<source address=/d' /etc/firewalld/zones/drop.xml
actionban = firewall-cmd --change-source=<ip> --zone=drop && firewall-cmd --change-source=<ip> --zone=drop --permanent
actionunban = firewall-cmd --remove-source=<ip> --zone=drop && firewall-cmd --remove-source=<ip> --zone=drop --permanent || echo 0

[Init]
```
`/etc/fail2ban/jail.local`
```
[DEFAULT]
banaction = custom-firewalld
```
`systemctl restart fail2ban`

**Comment by captainjs** [comment]
> So, trying this with current sshd-filter (0.10 and 0.9) show me:
> 
> ```
> fail2ban-regex 'Mar 25 18:09:29 SERVER sshd[1234]: Failed password for USER from 192.0.2.1 port 12345 ssh2' sshd
> ```
> 
> output of 0.9
> output of 0.10
> Because there are matches in both versions, I'd suggest - update your filter.
> For 0.9 version you can find it here - https://github.com/fail2ban/fail2ban/blob/0.9/config/filter.d/sshd.conf
> _(Click on button "Raw" to obtain raw copy)_

Hi @sebres,
1)  I have tried the same as you posted:
fail2ban-regex 'Mar 23 07:10:03 hostname sshd[1723479]: Failed password for root from 19x.x.x.x port 39722 ssh2' sshd
I get a match:
Failregex: 1 total
|-  #) [# of hits] regular expression
|   4) [1] ^Failed \b(?!publickey)\S+ for (?P<cond_inv>invalid user )?<F-USER>(?P<cond_user>\S+)|(?(cond_inv)(?:(?! from ).)*?|[^:]+)</F-USER> from <HOST>(?: (?:port \d+|on \S+)){0,2}(?: ssh\d*)?(?(cond_user): |(?:(?:(?! from ).)*)$)
`-
...
Lines: 1 lines, 0 ignored, 1 matched, 0 missed


Yet that IP is not found in the:
sudo fail2ban-client status ssh-iptables | grep 19x.x.x.x

Notes: 
-I do have currently like 75 ip banned there.
- I do have 61 Failed password, in a few hours/day on that same username, same ip:
- sudo less /var/log/secure | grep 19.x.x.x | grep Failed | wc -l 
61

I don't understand why it's not banned.

2) I don't understand this note in the filter comments:
# If you want to protect OpenSSH from being bruteforced by password
# authentication then get public key authentication working before disabling
# PasswordAuthentication in sshd_config.
Does that suggest that fail2ban won't protect against passwords bruteforce?
(well, it seems in the example above that you have showned that it's supposed to catch "Failed password for xxx"...)

Thanks

**Comment by sebres** [comment]
> Yet that IP is not found in the:
> `sudo fail2ban-client status ssh-iptables | grep 19x.x.x.x`

- is your jail calling really `ssh-iptables` (not `sshd`)? 
- it can be already unbanned, so not in status anymore (better check in fail2ban.log for `Found 19***` and `Ban 19***` )
- also take a look in [wiki :: How fail2ban works](https://github.com/fail2ban/fail2ban/wiki/How-fail2ban-works) for possible reasons if some IP does not found or not banned.

> `sudo less /var/log/secure | grep 19.x.x.x | grep Failed | wc -l` ==> 61

And what is the content of these lines?

> Does that suggest that fail2ban won't protect against passwords bruteforce?

No, it just suggest to disable password authentication. Fail2ban is and remain able to recognize password attempts.

**Comment by captainjs** [comment]
> > Yet that IP is not found in the:
> > `sudo fail2ban-client status ssh-iptables | grep 19x.x.x.x`
> 
> * is your jail calling really `ssh-iptables` (not `sshd`)?
ssh-iptables is the name of the section in the jail.local file:
[ssh-iptables]
enabled  = true
filter   = sshd
action   = iptables[name=SSH, port=ssh, protocol=tcp]
           sendmail-buffered[name=SSH, dest=email@xxx.com, sender=email@xxx.com]
logpath  = /var/log/secure
usedns   = no
maxretry = 10
bantime  = 604800
mode     = aggressive




> * it can be already unbanned, so not in status anymore (better check in fail2ban.log for `Found 19***` and `Ban 19***` )
True, but it happened to recently to be out of the ban. I have double checked carefully in the fail2ban.log.
sudo less /var/log/messages | grep action | grep 192.x.x.x return nothing. When I grep for action only, I see the numerous ban/unban, as per configuration.


> * also take a look in [wiki :: How fail2ban works](https://github.com/fail2ban/fail2ban/wiki/How-fail2ban-works) for possible reasons if some IP does not found or not banned.
Thanks I will read that, I already read a lot on the topic, I hope it will help me clarify also.

> > `sudo less /var/log/secure | grep 19.x.x.x | grep Failed | wc -l` ==> 61
> 
> And what is the content of these lines?
>
Ex:
Mar 23 07:19:54 hostname sshd[1764190]: Failed password for root from 192.x.x.x port 46278 ssh2
Mar 23 07:20:34 hostname sshd[1764413]: Failed password for root from 192.x.x.x port 53868 ssh2
Mar 23 07:21:20 hostname sshd[1764757]: Failed password for root from 192.x.x.x port 33232 ssh2
Mar 23 07:22:03 hostname sshd[1765598]: Failed password for root from 192.x.x.x port 40826 ssh2
Mar 23 07:22:51 hostname sshd[1766268]: Failed password for root from 192.x.x.x port 48418 ssh2
Mar 23 07:23:29 hostname sshd[1767507]: Failed password for root from 192.x.x.x port 56010 ssh2
Mar 23 07:24:15 hostname sshd[1767676]: Failed password for root from 192.x.x.x port 35370 ssh2
Mar 23 07:25:02 hostname sshd[1767882]: Failed password for root from 192.x.x.x port 42966 ssh2
Mar 23 07:26:44 hostname sshd[1768995]: Failed password for root from 192.x.x.x port 58176 ssh2
Mar 23 07:27:32 hostname sshd[1770140]: Failed password for root from 192.x.x.x port 37542 ssh2
Mar 23 07:28:18 hostname sshd[1771106]: Failed password for root from 192.x.x.x port 45136 ssh2
Mar 23 07:29:00 hostname sshd[1772062]: Failed password for root from 192.x.x.x port 52732 ssh2
Mar 23 07:30:30 hostname sshd[1772828]: Failed password for root from 192.x.x.x port 39690 ssh2
Mar 23 07:31:17 hostname sshd[1773194]: Failed password for root from 192.x.x.x port 47284 ssh2
Mar 23 07:32:09 hostname sshd[1773741]: Failed password for root from 192.x.x.x port 54884 ssh2
(The IP is identical each times)

I'm now trying to suspect that because it uses a different port, it might be considered as a unique attempt...


> > Does that suggest that fail2ban won't protect against passwords bruteforce?
> 
> No, it just suggest to disable password authentication. Fail2ban is and remain able to recognize password attempts.

Ok thanks for the clarification.

**Comment by sebres** [comment]
> ssh-iptables is the name of the section in the jail.local file:

Why? Why you don't use default jail which is already there (and just overwrites the parameters you need)?...

Anyway, you have a mistake here - mode `aggressive` will not work in this way - per default the filter is set like here:
https://github.com/fail2ban/fail2ban/blob/e587526ede759977b8b483deeaa644f45f97cea3/config/jail.conf#L129
but you overwrite it (without mode), so it works with mode `normal`. If you still need your custom jail you must do it this way:
```diff
 [ssh-iptables]
-filter = sshd
+filter = sshd[mode=%(mode)s]
+# or directly - filter = sshd[mode=aggressive]
 ...
 mode = aggressive
```
then it will really supply the jail parameter to filter.

> `Mar 23 07:19:54 hostname sshd[1764190]: Failed password for root from 192.x.x.x port 46278 ssh2`

If I replace `192.x.x.x` with `192.0.2.1` in your excerpt and try `fail2ban-regex` over this excerpt with sshd filter (no matter as `sshd` or `sshd[mode=aggressive]`, which is not really needed here), I get 15 matches as expected:
`Lines: 15 lines, 0 ignored, 15 matched, 0 missed`

So either your filter is too old (no `Found` at all in fail2ban.log), which were strange because you wrote "I get a match" in your previous comment. 
Or the IP is in ignored IPs (`ignoreip`, `ignoreself`, `ignorecommand`, etc) or the IP makes fewer failures than your `maxretry` (within your `findtime`). Or you have some basic issue, like wrong timezone (the time in log like `07:19:54` is incorrect e. g. too old and outside of `now - findtime`), or something else. 
I can only repeat, read the wiki attentively.

**Comment by captainjs** [comment]
Hi @sebres,
I have created a paste for you:

https://paste.centos.org/view/6492dcb3

I also got a match using the regex. And usually it works, some connections are banned.
I have verified, in the jail.local there is only 3 ignore ip (2subnets and 1 ip):
ignoreip  = 127.0.0.1/24 10.0.0.0/8 155.x.x.x (I replaced that last ip with x).

I have read the wiki attentively, maybe I learn a thing or two, I'm not closer to find why sometimes it bans some ips and sometimes not...

I verified the timestamps and they occured fairly recently. Other actions which resulted in a ban (again it's all timed) also occured fairly before the ban.

findtime is by default (10min I think based on wiki) and maxretry is set to 10.

Thanks,
PS: I think it start to be not related to this issue anymore, as initially I though it was not catching "Failed password".
What is the best way to communicate on that?
I went to IRC, I rarely get any answers after days.

**Comment by sebres** [comment]
As for "Failed password" and `maxretry = 10` within `findtime = 10m` - in your excerpt I see for example first 10 messages ("Failed password ... from 192.x.x.x"), but they takes place in 15 and half minutes `16:29:49 - 16:14:17 == 15:32`. 
```
Mar 23 16:14:17 serverhostname sshd[2146711]: Failed password ...
Mar 23 16:15:45 serverhostname sshd[2148227]: Failed password ...
...6 similar messages...
Mar 23 16:27:38 serverhostname sshd[2156915]: Failed password ...
Mar 23 16:29:49 serverhostname sshd[2158637]: Failed password ...
```
It is not enough to cause a ban for this IP, so you have either to increase findtime or decrease maxretry.
And messages "Disconnected from invalid user" don't matter here, because as already said above due to mentioned mistake aggressive mode was not enabled in your case.

> I verified the timestamps and they occured fairly recently. 

So this can exclude the time-zone issue, if you mean that 16:29:49 is the real time its occurs on the side of sshd logging.
But what is about fail2ban? 
Also note `Found` != `Ban`, so you *must* see corresponding message `[ssh-iptables] Found 192.x.x.x` in your fail2ban.log (e. g. 10 times between 16:14 and 16:29). Don't you?

> PS: I think it start to be not related to this issue anymore, as initially I though it was not catching "Failed password".
> What is the best way to communicate on that?

Indeed. 
You could open new issue, but it would not lead to solution (and probably will only waste our time) unless you can demonstrate that some IP despite of each reason, mentioned in the wiki FAQ I provided above, would be not found or not banned.
Neither I see this (well basically I'm seeing vice versa proofs at the moment), nor your grepped excerpts are appropriate for that, for instance one doesn't see the counterpart of fail2ban.log, because you grep only for action (so see Bans only).
Check a full excerpt of fail2ban.log for time between 16:14 and 16:30, whether you'd see some findings for this IP, or some errors or whatever else.
On linux one made mostly `tail -f  /var/log/secure /var/log/fail2ban.log` and monitors both logs at once parallel (so you have a chance to consider how fail2ban reacts on related secure messages).
If it is too wordy you can make it fewer verbose, e. g. using grep but make the pattern more precise bit by bit, for example:
```bash
# show only "Failed password" and everything from fail2ban:
tail -f  /var/log/secure /var/log/fail2ban.log | grep -E 'Failed password|fail2ban'
# show only "Failed password" and everything from fail2ban filters (e. g. whether it find something):
tail -f  /var/log/secure /var/log/fail2ban.log | grep -E 'Failed password|fail2ban\.filter'
# show only "Failed password" and everything from fail2ban filters and actions:
tail -f  /var/log/secure /var/log/fail2ban.log | grep -E 'Failed password|fail2ban\.(filter|action)'
```

**Comment by captainjs** [comment]
> As for "Failed password" and `maxretry = 10` within `findtime = 10m` - in your excerpt I see for example first 10 messages ("Failed password ... from 192.x.x.x"), but they takes place in 15 and half minutes `16:29:49 - 16:14:17 == 15:32`.

Hi @sebres thanks. I did not understand why an IP was not banned.
Like you explained there's also the "findtime" setting (that someone set in my case to findtime=100 in the [default] section).
I carefully looked at tested and indeed, if the "Failed Password" entries are not within the time period of "now-findtime" then it's not banned (Note: like you said, also those entries should be higher than the maxretry number). 
I only looked at the maxretries in the past, and since I was doing the Failed attempt manually, I did not check findtime and I was always within that time range.

That explains the situation to me, I will fix the configuration. Probably a findtime of 1800 seems reasonnable, in combination with maxretry=10. That means if someone has 10 Failed password within 30min, they get banned. 

Cheers,

**Comment by sebres** [comment]
Oh God!.. 
Why we're marking an effort to write and maintain the docu and wiki, if nobody reads them?..

I referenced several times the [wiki](https://github.com/fail2ban/fail2ban/wiki/How-fail2ban-works)... Here is direct quote from first A to first Q there (almost at begin):
- the IP goes to ban if it makes at least `maxretry` failures within `findtime` seconds. So if you've configured `maxretry=5` and `findtime=10m` (default values) then it needs at least 5 failures (5 attempts) within 10 minutes to ban an IP.<br/>
Each failure (attempt) will be logged in `fail2ban.log` as:<br/>
`  INFO    [jail] Found 192.0.2.25`<br/>
First if you'll see at least 5 such lines with this IP address within 10 minutes, the IP goes banned and you should see:<br/>
`  NOTICE  [jail] Ban 192.0.2.25`<br/>

**Comment by captainjs** [comment]
Hi @sebres,
I read the wiki several times, after you referred to it several times. Thanks again for sharing and orientating me there.

I did found the explanation there, I tried to make a some sort of summary of why the "Failed password" was not resulting in a ban. Since you also complemented the information with the timestamps and shown me that actually the period I was looking for was long, I made some sort of "summary" that could be helpful for anyone reading the topic and see the issue I faced and the quick solution... Indeed I should/could have mentionned the wiki on github also (since I also used it to understand the issue).

Your efforts are not in vain, the wiki in Github is useful and yes, since there are a lot of information out there, I did miss that one during research. Also, the initial wiki I had found, was different than the one you shared:(https://www.fail2ban.org/wiki/index.php/Main_Page).

It's a lot of information and easy to get lost.
